query-translator/tests/translate.rs

```diff
@@ -741,28 +741,91 @@ fn test_fulltext() {
+    assert_eq!(sql, "SELECT DISTINCT `datoms01`.e AS `?entity`, \
+                                     `fulltext_values00`.text AS `?value`, \
+                                     `datoms01`.tx AS `?tx` \
+                     FROM `fulltext_values` AS `fulltext_values00`, \
+                          `datoms` AS `datoms01`, \
+                          `datoms` AS `datoms02` \
+                     WHERE `datoms01`.a = 100 \
+                       AND `datoms01`.v = `fulltext_values00`.rowid \
+                       AND `fulltext_values00`.text MATCH $v0 \
+                       AND `datoms02`.a = 99 \
+                       AND `datoms02`.v = 0 \
+                       AND `datoms01`.e = `datoms02`.e");
     assert_eq!(args, vec![make_arg("$v0", "needle"),]);
 
     let query = r#"[:find ?entity ?value ?tx :where [?entity :foo/bar ?score] [(fulltext $ :foo/fts "needle") [[?entity ?value ?tx ?score]]]]"#;
     let SQLQuery { sql, args } = translate(&schema, query);
-    assert_eq!(sql, "SELECT DISTINCT `datoms00`.e AS `?entity`, `fulltext_values01`.text AS `?value`, `datoms02`.tx AS `?tx` FROM `datoms` AS `datoms00`, `fulltext_values` AS `fulltext_values01`, `datoms` AS `datoms02`, (SELECT 0 AS `?score` WHERE 0 UNION ALL VALUES (0)) AS `c00` WHERE `datoms00`.a = 99 AND `datoms02`.a = 100 AND `datoms02`.v = `fulltext_values01`.rowid AND `fulltext_values01`.text MATCH $v0 AND `datoms00`.e = `datoms02`.e AND `datoms00`.v = `c00`.`?score`");
+    assert_eq!(sql, "SELECT DISTINCT `datoms00`.e AS `?entity`, \
+                                     `fulltext_values01`.text AS `?value`, \
+                                     `datoms02`.tx AS `?tx` \
+                     FROM `datoms` AS `datoms00`, \
+                          `fulltext_values` AS `fulltext_values01`, \
+                          `datoms` AS `datoms02` \
+                     WHERE `datoms00`.a = 99 \
+                       AND `datoms02`.a = 100 \
+                       AND `datoms02`.v = `fulltext_values01`.rowid \
+                       AND `fulltext_values01`.text MATCH $v0 \
+                       AND `datoms00`.v = 0 \
+                       AND `datoms00`.e = `datoms02`.e");
     assert_eq!(args, vec![make_arg("$v0", "needle"),]);
 }
+
+#[test]
+fn test_fulltext_inputs() {
+    let schema = prepopulated_typed_schema(ValueType::String);
+
+    // Bind ?entity. We expect the output to collide.
+    let query = r#"[:find ?val
+                    :in ?entity
+                    :where [(fulltext $ :foo/bar "hello") [[?entity ?val _ _]]]]"#;
+    let SQLQuery { sql, args } = translate(&schema, query);
+    assert_eq!(sql, "SELECT DISTINCT `fulltext_values00`.text AS `?val` \
```

Comment:
Is this test correct?  This doesn't seem to be any different from the query without `:in ?entity`.   Can you spell out what's changed here, or compare the two (with and without the `:in`) in the test?